File: 1518. Water Bottles.cpp

class Solution
{
public:
    int numWaterBottles(int numBottles, int numExchange)
    {

        int last = numBottles;
        int sum = numBottles;
        int a = sum;
        int shes = 0;
        while (last > 0)
        {
            last = last / numExchange;
            sum = sum + last;
            shes = a % numExchange;

            last = shes + last;
            a = last;
            if (last < numExchange)
            {
                break;
            }
        }
        return sum;
    }
};

File: leetcode /1518. Water Bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:

        sum: int = numBottles
        last: int = numBottles
        reminder: int = 0
        while True:
            last = int(numBottles / numExchange)  # 1
            sum += last
            print(sum)

            reminder = numBottles % numExchange  # 2
            numBottles = last + reminder

            if numBottles < numExchange:
                break

        return int(sum)

#-----------------------------------------------------------------------

e1 = Solution()
ans = e1.numWaterBottles(15, 4)
print(ans)

